### ISSUE TITLE
TimeZoneSerializerField not accepting choices argument

### ISSUE THREAD

**Issue by gabn88** [mfogel/django-timezone-field#69]
And as a result, the OPTIONS request does not show all the possible choices in the API.

This is related to https://github.com/mfogel/django-timezone-field/issues/61

In my opinion, this should be the first step.

**Comment by gabn88** [comment]
```
class TimeZoneField(serializers.ChoiceField):
    def __init__(self, **kwargs):
        super().__init__(TimeZoneField_.default_choices, **kwargs)

    def to_representation(self, value):
        return str(super().to_representation(value))
```
This is how I do it now. Maybe this could be incorporated into this package.

**Comment by mfogel** [comment]
Agreed it would be great to respond to an `OPTIONS` request with the list of possible choices. I didn't realize that was provided by django rest framework choice field.

@gabn88 would it be possible to expand the field class you have there to accept an optional `choices` argument? The idea would be to support the case where someone wants the serializer field to only display a subset of all possible timezone choices.

I think if that was added to this implementation, and then also some basic tests, we could merge it in named something like `TimeZoneSerializerChoiceField`. Then it could live in parallel with the currently-existing `TimeZoneSerializerField` which we would need to keep around for backwards-compatibility till at least the next major version bump.

**Comment by hyshka** [comment]
Thanks for this. I was using a different implementation with v4.0 which broke when I upgraded to v4.1.2. This fixed it for me.

![image](https://user-images.githubusercontent.com/2090758/111820953-b73bd600-88a7-11eb-9cc5-b3b5e4804732.png)

**Comment by normic** [comment]
> ```
> class TimeZoneSerializerField(serializers.ChoiceField):
>     def __init__(self, **kwargs):
>         super().__init__(TimeZoneField.default_choices, **kwargs)
> 
>     def to_representation(self, value):
>         return str(super().to_representation(value))
> ```
> 
> This is how I do it now. Maybe this could be incorporated into this package. It now shows a list of all possible timezone fields in the browsable API and in the in an OPTIONS request.

@gabn88 How are you dealing with this now? As far as I've seen, there is no .default_choices anymore. Or do you stick with an old version?

**Comment by gabn88** [comment]
@normic I use it like this
```
class TimeZoneSerializerChoiceField(TimeZoneSerializerField, serializers.ChoiceField):
    def __init__(self, **kwargs):
        super().__init__([(tz, tz) for tz in self.tz_backend.base_tzstrs], **kwargs)

```

**Comment by marcomartinscastanho** [comment]
Another reason why `TimeZoneSerializerField` should be a ChoiceField is that, if you currently use a OpenAPI library like `drf-spectacular`, it crashes trying to determine the choices.

I'm doing this (using `zoneinfo`, not pytz):
```py
import zoneinfo
from timezone_field.rest_framework import TimeZoneSerializerField as DefaultTimeZoneSerializerField


class TimeZoneSerializerField(DefaultTimeZoneSerializerField, serializers.ChoiceField):
    def __init__(self, **kwargs):
        super().__init__(choices=zoneinfo.available_timezones(), **kwargs)

```
